modules/gdpr_tasks/src/Traversal/RightToBeForgottenEntityTraversal.php

```diff
@@ -0,0 +1,232 @@
+      $results['successes'][] = $entity;
+    }
+    else {
+      $results['failures'][] = $entity;
+    }
+  }
+
+  /**
+   * Removes the field value.
+   *
+   * @param \Drupal\Core\Field\FieldItemListInterface $field
+   *   The current field to process.
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity to remove.
+   *
+   * @return array
+   *   First element is success boolean, second element is the error message.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  private function remove(FieldItemListInterface $field, EntityInterface $entity) {
+    try {
+      // If this is the entity's ID, treat the removal as remove the entire
+      // entity.
+      $entity_type = $entity->getEntityType();
+      if ($entity_type->getKey('id') == $field->getName()) {
+        $entity->delete();
+      }
+      // Check if the property can be removed.
+      elseif (!GDPRCollector::propertyCanBeRemoved($entity_type, $field->getFieldDefinition(), $error_message)) {
+        return [FALSE, $error_message];
+      }
+
+      // Otherwise assume we can simply clear the field.
+      $field->setValue(NULL);
+      return [TRUE, NULL];
+    }
+    catch (ReadOnlyException $e) {
+      return [FALSE, $e->getMessage()];
+    }
+  }
+
+  /**
+   * Runs anonymize functionality against a field.
+   *
+   * @param \Drupal\Core\Field\FieldItemListInterface $field
+   *   The field to anonymize.
+   * @param \Drupal\Core\Field\FieldDefinitionInterface $field_definition
+   *   The field definition.
+   * @param \Drupal\gdpr_fields\Entity\GdprField $field_config
+   *   GDPR field configuration.
+   *
+   * @return array
+   *   First element is success boolean, second element is the error message.
+   */
+  private function anonymize(FieldItemListInterface $field, FieldDefinitionInterface $field_definition, GdprField $field_config) {
+    $anonymizer_id = $this->getAnonymizerId($field_definition, $field_config);
+
+    if (!$anonymizer_id) {
+      return [
+        FALSE,
+        "Could not anonymize field {$field->getName()}. Please consider changing this field from 'anonymize' to 'remove', or register a custom anonymizer.",
+        NULL,
+      ];
+    }
+
+    try {
+      $anonymizer = $this->anonymizerFactory->get($anonymizer_id);
+      $field->setValue($anonymizer->anonymize($field->value, $field));
+      return [TRUE, NULL, $anonymizer_id];
+    }
+    catch (\Exception $e) {
+      return [FALSE, $e->getMessage(), NULL];
+    }
+  }
+
+  /**
+   * Gets the ID of the anonymizer plugin to use on this field.
+   *
+   * @param \Drupal\Core\Field\FieldDefinitionInterface $field_definition
+   *   The field definition.
+   * @param \Drupal\gdpr_fields\Entity\GdprField $field_config
+   *   GDPR field configuration.
+   *
+   * @return string
+   *   The anonymizer ID or null.
+   */
+  private function getAnonymizerId(FieldDefinitionInterface $field_definition, GdprField $field_config) {
```

Comment:
Is `private` suitable, just wondering how likely this is to extended...

Reply:
Probably `private` is better for that very reason :P

Reply:
I'd say private as we already provide ways to get the anonymizer (either via plugin or `gdpr_type_anonymizers` hook)